libsmtutil/CHCSmtLib2Interface.cpp

```
@@ -203,12 +206,14 @@ string CHCSmtLib2Interface::querySolver(string const& _input)
 	util::h256 inputHash = util::keccak256(_input);
 	if (m_queryResponses.count(inputHash))
 		return m_queryResponses.at(inputHash);
-	if (m_smtCallback)
+
+	if ((m_enabledSolvers.smtlib2 || m_enabledSolvers.eld) && m_smtCallback)
```

maybe unnecessary in the current version to test which solvers were enabled here

---

If the callback is not allowed for other solvers, shouldn't we report an error instead of ignoring it?

---

it's not that it's not allowed, it's just not used

---

Actually, since this is CHC, we could add an assertion that the first arg of the conjunction should be true. What do you think @cameel ?

---

Ah, I didn't notice that it's guaranteed to be true. Sure, assert sounds right in that case.